### ISSUE TITLE
Implement rewrite(Add)

### ISSUE THREAD

**Issue by Davide-sd** [gutow/Algebra_with_Sympy#8]
Hello gutow, this is a very nice package.

Would it be possible to implement the `rewrite` method so that when a user execute `eq.rewrite(Add)` it returns `eq.lhs - eq.rhs`, like SymPy's `Eq` does?

**Comment by gutow** [comment]
> Hello gutow, this is a very nice package.

Thank you.

**Comment by gutow** [comment]
@Davide-sd should this issue remain open? Is there another use for `rewrite` other then making input for `solve`?

**Comment by Davide-sd** [comment]
Please leave it open for a few more days, as I'm still trying this module and I have to fully understand if this feature might be useful or what...

**Comment by Davide-sd** [comment]
@gutow : how do you usually run tests on your local machine? I naively run the `pytest` command from the root folder, but I got the following error:

```
_________________ERROR collecting tests/test_algebraic_equation.py _________________________________________
tests/test_algebraic_equation.py:55: in <module>
    exec(execstr, globals(), locals())
<string>:1: in <module>
    ???
E   TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

**Comment by gutow** [comment]
> @gutow : how do you usually run tests on your local machine? I naively run the `pytest` command from the root folder, but I got the following error:
> 
> ```
> _________________ERROR collecting tests/test_algebraic_equation.py _________________________________________
> tests/test_algebraic_equation.py:55: in <module>
>     exec(execstr, globals(), locals())
> <string>:1: in <module>
>     ???
> E   TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
> ```
Davide,
Sorry about that. In my rush to get out the fix to be compatible with sympy 1.11 (added a function that isn't part of function class), I did not update the way tests load and override the sympy functions.

I am working on a more robust way of doing this that should not require updating tests when sympy adds other non-compliant functions.

If you need to run tests you can fix this by adding the string `'piecewise_exclusive'` to `skip` list that begins on line 49 of test_algebraic_equation.py.
